src/main/java/com/google/jenkins/plugins/computeengine/NoDelayProvisionerStrategy.java

```diff
@@ -0,0 +1,109 @@
+package com.google.jenkins.plugins.computeengine;
+
+import hudson.Extension;
+import hudson.model.Label;
+import hudson.model.LoadStatistics;
+import hudson.slaves.Cloud;
+import hudson.slaves.CloudProvisioningListener;
+import hudson.slaves.NodeProvisioner;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+
+/**
+ * Implementation of {@link NodeProvisioner.Strategy} which will provision a new node immediately as
+ * a task enter the queue.
+ */
+@Extension(ordinal = 100)
+public class NoDelayProvisionerStrategy extends NodeProvisioner.Strategy {
+
+  private static final Logger LOGGER = Logger.getLogger(NoDelayProvisionerStrategy.class.getName());
+
+  private static final boolean DISABLE_NODELAY_PROVISING =
+      Boolean.valueOf(
+          System.getProperty(
+              "com.google.jenkins.plugins.computeengine.disableNoDelayProvisioning"));
+
+  @Override
+  public NodeProvisioner.StrategyDecision apply(NodeProvisioner.StrategyState strategyState) {
+    if (DISABLE_NODELAY_PROVISING) {
+      LOGGER.log(Level.FINE, "Provisioning not complete, NoDelayProvisionerStrategy is disabled");
+      return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+    }
+    final Label label = strategyState.getLabel();
+
+    LoadStatistics.LoadStatisticsSnapshot snapshot = strategyState.getSnapshot();
+    int availableCapacity =
+    snapshot.getAvailableExecutors()   // live executors
+            + snapshot.getConnectingExecutors()  // executors present but not yet connected
+            + strategyState.getPlannedCapacitySnapshot()     // capacity added by previous strategies from previous rounds
+            + strategyState.getAdditionalPlannedCapacity();  // capacity added by previous strategies _this round_
+    int currentDemand = snapshot.getQueueLength();
+    LOGGER.log(
+        Level.FINE,
+        "Available capacity={0}, currentDemand={1}",
+        new Object[] {availableCapacity, currentDemand});
+    if (availableCapacity < currentDemand) {
+      List<Cloud> jenkinsClouds = new ArrayList<>(Jenkins.get().clouds);
+      Collections.shuffle(jenkinsClouds);
+      for (Cloud cloud : jenkinsClouds) {
+        int workloadToProvision = currentDemand - availableCapacity;
+        if (!(cloud instanceof ComputeEngineCloud)) continue;
+        if (!cloud.canProvision(label)) continue;
+        ComputeEngineCloud gcp = (ComputeEngineCloud) cloud;
+        if (!gcp.isNoDelayProvisioning()) continue;
+        for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+          if (cl.canProvision(cloud, strategyState.getLabel(), workloadToProvision) != null) {
+            continue;
```

Comment:
This continue applies to the inner loop. What then is the intended effect of the loop?

Reply:
That's caused by [JENKINS-60055](https://issues.jenkins-ci.org/browse/JENKINS-60055) and that's the default behaviour in the NodeProvisioner class too:
- https://github.com/jenkinsci/jenkins/blob/450d87491df16598409d26511bcad02978eaa8bb/core/src/main/java/hudson/slaves/NodeProvisioner.java#L721-L726